nova/modules/core/libraries/Nova_extension.php

```diff
@@ -0,0 +1,57 @@
+<?php if ( ! defined('BASEPATH')) exit('No direct script access allowed');
+/**
+ * Container for accessing the {@see Nova_extension_container} objects
+ * instantiated for each loaded extension. These can be accessed such as:
+ *
+ *    $this->extension['jquery']
+ *
+ * Objects attached to an extension can then be accessed such as:
+ *
+ *    $this->extension['jquery']['generator']
+ *
+ * Other methods such as {@see Nova_extension_container::view()} can also
+ * be called this way, such as:
+ *
+ *    $this->extension['my_extension']
+ *         ->location('my_extension_view',$this->skin,'main',$data)
+ *
+ * This will return the view
+ * application/extensions/my_extension/views/main/pages/my_extension_view.php
+ *
+ * @package		Nova
+ * @category	Library
+ * @author		Anodyne Productions
+ * @copyright	2015 Anodyne Productions
+ */
+
+abstract class Nova_extension implements ArrayAccess {
+  
+  public $facades = array();
```

Comment:
Let's try to stay consistent with the use of the array short syntax. We can switch this over to `[]`.

Reply:
Why are we using the term `facades` here?

Reply:
I was using the term `facade` because they're frontends to functionality supported by the extension. We could go with something else like `libraries` or `attached` if you'd prefer.

I will fix to using `[]`. I just noticed some places in the codebase where `array()` was used, so I just emulated that.